test/TestRist.cpp

```diff
@@ -0,0 +1,329 @@
+
+    // check for sender connecting
+    {
+        std::unique_lock<std::mutex> lock(connectedMutex);
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(10), [&]() { return connected; });
+        ASSERT_TRUE(successfulWait) << "Timeout waiting for sender to connect";
+    }
+
+    rist_peer* client;
+    receiver.getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 1);
+            client = activeClients.begin()->first;
+        });
+
+    RISTNetSender sender2;
+    ASSERT_TRUE(sender2.initSender(senderInterfaces, senderSettings));
+
+    // check for sender2 connecting
+    {
+        std::unique_lock<std::mutex> lock(connectedMutex);
+        connected = false;
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(10), [&]() { return connected; });
+        ASSERT_TRUE(successfulWait) << "Timeout waiting for sender2 to connect";
+    }
+
+    EXPECT_TRUE(receiver.closeClientConnection(client));
+    {
+        std::unique_lock<std::mutex> lock(disconnectMutex);
+        bool successfulWait =
+            disconnectCondition.wait_for(lock, std::chrono::seconds(12), [&]() { return disconnected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for sender disconnect";
+    }
+
+    receiver.closeAllClientConnections();
+    {
+        std::unique_lock<std::mutex> lock(disconnectMutex);
+        disconnected = false;
+        bool successfulWait =
+            disconnectCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return disconnected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for sender2 disconnect";
+    }
+}
+
+TEST(TestRist, SendReceive) {
+    RISTNetReceiver receiver;
+    uint16_t nReceivedPackets = 0;
+    const uint16_t kSentPackets = 5;
+    const uint16_t kBufferSize = 1024;
+    std::vector<std::string> receiverInterfaces{"rist://@0.0.0.0:8000"};
+    RISTNetReceiver::RISTNetReceiverSettings receiverSettings;
+    rist_peer_config_defaults_set(&receiverSettings.mPeerConfig);
+    receiverSettings.mPSK = kValidPsk;
+    auto receiverCtx = std::make_shared<RISTNetReceiver::NetworkConnection>();
+
+    receiver.validateConnectionCallback = [&](const std::string& ipAddress, uint16_t port) { return receiverCtx; };
+
+    std::condition_variable receiverCondition;
+    std::mutex receiverMutex;
+    receiver.networkDataCallback = [&](const uint8_t* buf, size_t size,
+                                       std::shared_ptr<RISTNetReceiver::NetworkConnection>& connection, rist_peer* peer,
+                                       uint16_t connectionId) {
+        EXPECT_EQ(connection, receiverCtx);
+        EXPECT_EQ(size, kBufferSize);
+        {
+            std::lock_guard<std::mutex> lock(receiverMutex);
+            EXPECT_EQ(*buf, nReceivedPackets + '0');
```

Comment:
Might be more readable if you flip the operands, '0' + nReceivedPackets. Same below in the sending code.